dftimewolf/lib/collectors/gsheets.py

```diff
@@ -0,0 +1,243 @@
+# -*- coding: utf-8 -*-
+"""Pulls entries from Google Sheets."""
+
+import os.path
+import re
+import tempfile
+from typing import List, Optional, TYPE_CHECKING
+
+from pandas.core.frame import DataFrame
+
+from dftimewolf.lib import module
+from dftimewolf.lib.containers import containers
+from dftimewolf.lib.modules import manager as modules_manager
+import filelock
+from google.auth.exceptions import DefaultCredentialsError
+from google.auth.exceptions import RefreshError
+from google.auth.transport.requests import Request
+from google.oauth2.credentials import Credentials
+from google_auth_oauthlib.flow import InstalledAppFlow
+from googleapiclient import discovery
+import pandas as pd
+
+if TYPE_CHECKING:
+  from dftimewolf.lib import state
+
+
+class GoogleSheetsCollector(module.BaseModule):
+  """Collector for entries from Google Sheets. """
+
+  SCOPES = ['https://www.googleapis.com/auth/spreadsheets.readonly']
+
+  _CREDENTIALS_FILENAME = '.dftimewolf_google_sheets_credentials.json'
+  _CLIENT_SECRET_FILENAME = '.dftimewolf_google_sheets_client_secret.json'
+
+  def __init__(self,
+               state: 'state.DFTimewolfState',
+               name: Optional[str] = None,
+               critical: bool = False) -> None:
+    """Initializes a Google Sheets collector."""
+    super(GoogleSheetsCollector, self).__init__(
+        state, name=name, critical=critical)
+    self._credentials = None
+    self._spreadsheet_id = ''
+    self._sheets_names = ''
+    self._mandatory_columns = ['message', 'datetime', 'timestamp_desc']
```

Comment:
Please add a comment here indicating that these are mandatory columns for a succesful timesketch export.

Reply:
Done